contracts/staking/slashing/SlashingManager.sol

```diff
@@ -0,0 +1,126 @@
+pragma solidity ^0.5.2;
+
+import {Ownable} from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import {SafeMath} from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import {RLPReader} from "solidity-rlp/contracts/RLPReader.sol";
+import {IERC20} from "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+
+import {StakeManager} from "../stakeManager/StakeManager.sol";
+import {Registry} from "../../common/Registry.sol";
+import {StakingInfo} from "../StakingInfo.sol";
+import "./ISlashingManager.sol";
+
+
+contract SlashingManager is ISlashingManager, Ownable {
+    using SafeMath for uint256;
+    using RLPReader for bytes;
+    using RLPReader for RLPReader.RLPItem;
+
+    modifier onlyStakeManager() {
+        require(registry.getStakeManagerAddress() == msg.sender);
+        _;
+    }
+
+    constructor(address _registry, address _logger) public {
```

Comment:
Please accept `_heimdallId` as an initialization param.

Reply:
reading from root chain is better 🤔💭.

Reply:
You are not reading from RootChain either, you are inheriting a hardcoded value from `ISlashingManager`.

Reply:
oh💩 lost that change, added into the initialization param.